app/assets/javascripts/standalone/facility_teleconsult_fields.js

```diff
@@ -0,0 +1,74 @@
+FacilityTeleconsultFields = function () {
+  this.userSearch =  new UserTeleconsultSearch();
+
+  this.toggleTeleconsultationFields = () => {
+    $("#teleconsultation-fields").toggle();
+  }
+
+// Adding/removing MO from search results
+  this.medicalOfficerCard = (user) => {
+    let card = $("template#medical-officer-card").html();
+    let $card = $(card);
+    let userID = user["userId"];
+
+    $card.find(".medical-officer-id").val(userID);
+    $card.find(".medical-officer-name").html(user["userFullName"]);
+    $card.find(".medical-officer-registration-facility").html(user["userRegistrationFacility"]);
+    $card.find(".medical-officer-phone-number").html(user["userTeleconsultationPhoneNumber"]);
+    $card.find("[data-user-id]").attr("data-user-id", userID);
+    $card.attr("data-user-id", userID);
+    $card.find("a").attr("href", `/admin/users/${userID}/edit`); //TODO: rewrite
+
+    return $card;
+  }
+
+  this.existingUsers = () => {
+    return $(".medical-officer-card").map((_, el) => $(el).attr("data-user-id")).get();
+  }
+
+  this.isAdded = (user) => {
+    let addedMedicalOfficerIDs = this.existingUsers();
+    return addedMedicalOfficerIDs.includes(user["userId"])
+  }
+
+  this.medicalOfficersCount = () => {
+    return $(".medical-officer-card").length
+  }
+
+  this.hideNoMedicalOfficers = () => {
+    $(".no-medical-officers").hide()
+  }
+
+  this.showNoMedicalOfficers = () => {
+    $(".no-medical-officers").show()
+  }
+
+  this.addMedicalOfficer = (user) => {
+    if (this.isAdded(user)) return false;
+
+    $(".medical-officers").append(this.medicalOfficerCard(user));
+    this.hideNoMedicalOfficers();
+  }
+
+  this.removeMedicalOfficer = (userID) => {
+    $(`[data-user-id="${userID}"]`).remove();
+
+    if (!this.medicalOfficersCount()) this.showNoMedicalOfficers();
+  }
+
+  $(document).ready(() => {
+    let teleconsultationFields = this;
+
+    // Populate MOs from search results
+    $("body").on("click", ".mo-search .typeahead-dropdown-row", function () {
+      let user = $(this).data();
+      teleconsultationFields.addMedicalOfficer(user);
+      teleconsultationFields.userSearch.emptyTypeahead();
+    });
+
+    $("body").on("click", ".remove-medical-officer", function () {
+      let userID = $(this).attr('data-user-id');
+      teleconsultationFields.removeMedicalOfficer(userID);
+    });
```

Comment:
The classes look great. Could we place these JS ditties back in `common` so they're part of a unified JS bundle for the app? If we can introduce a convenient interface to this class, we can then move the _initialization_ (which we don't want to do on every page) to a tiny snippet of inline JS on the page itself. For example:

```html
blah blah erb markup blah blah

<script>
  facilityTeleconsultFields = new FacilityTeleconsultFields();
  facilityTeleconsultFields.listen();
</script>
```

```js
  // blah blah teleconsultfieldsstuff
  this.listen() {
        let teleconsultationFields = this;

    // Populate MOs from search results
    $("body").on("click", ".mo-search .typeahead-dropdown-row", function () {
      let user = $(this).data();
      teleconsultationFields.addMedicalOfficer(user);
      teleconsultationFields.userSearch.emptyTypeahead();
    });

    $("body").on("click", ".remove-medical-officer", function () {
      let userID = $(this).attr('data-user-id');
      teleconsultationFields.removeMedicalOfficer(userID);
    });
  }
```

See #1297 for inspiration 😄

Reply:
This was done in 6dfa50e.

Reply:
Lol, did I review an outdated diff 🤦 

Neato. I still like the idea of wrapping the event listener registration in a method, rather than letting them implicitly run on initialization. We could even run them explicitly on registration inside a constructor method. Shall we do that?

Reply:
Sounds good 💯